addons/mail/static/src/models/thread_cache/thread_cache.js

```diff
@@ -163,7 +163,7 @@ function factory(dependencies) {
                     message.id > this.lastFetchedMessage.id
                 );
             }
-            messages = messages.concat(newerMessages);
+            messages = messages.concat(newerMessages, this.senderPendingMessagesToBeSent, this.senderErrorNotDelivred);
```

Comment:
Don't use the related fields in business code, prefer following the relations explicitly.
Just keep related if you need them for dependencies.